reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+
+The lifetime of the profiling storage in seconds. The data will be deleted when
+the lifetime is expired. It becomes the service container parameter called
+``profiler.storage.lifetime``.
+
+
+matcher
+.......
+
+.. seealso::
+
+    See :doc:`/cookbook/profiler/matchers` for more information about using
+    matchers to enable/disable the profiler.
+
+ip
+""
+
+**type**: ``string``
+
+If set, the profiler will only be enabled when the current IP address matches.
+
+.. _reference-profiler-matcher-path:
+
+path
+""""
+
+**type**: ``string``
+
+If set, the profiler will only be enabled when the current path matches.
+
+service
+"""""""
+
+**type**: ``string``
+
+This setting contains the service id of a custom matcher.
+
+router
+~~~~~~
+
+resource
+........
+
+**type**: ``string`` **required**
+
+Specifies the path to the routes used by the default router.
+
+It becomes the service container parameter called ``router.resource``.
+
+type
+....
+
+**type**: ``string``
+
+The type of the resource to hint the loaders about the format. This isn't
+needed when you use the default routers with the expected file extensions
+(``.xml``, ``.yml`` / ``.yaml``, ``.php``).
+
+http_port
+.........
+
+**type**: ``integer`` **default**: ``80``
+
+The port for normal http requests (this is used when matching the scheme).
+
+It becomes the service container parameter called ``router.resource``.
 
-    .. code-block:: xml
+https_port
+..........
 
-        <!-- app/config/config.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:framework="http://symfony.com/schema/dic/symfony"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+**type**: ``integer`` **default**: ``443``
 
-            <framework:config trusted-proxies="192.0.0.1, 10.0.0.0/8" />
-        </container>
+The port for https requests (this is used when matching the scheme).
 
-    .. code-block:: php
+strict_requirements
+...................
 
-        // app/config/config.php
-        $container->loadFromExtension('framework', array(
-            'trusted_proxies' => array('192.0.0.1', '10.0.0.0/8'),
-        ));
+**type**: ``mixed`` **default**: ``true``
 
-.. _reference-framework-form:
+Determines the behaviour when a route matches, but the parameters do not match
+the specified requirements for that route. Can be one of:
 
-form
-~~~~
+``true``
+    Throw an exception when the requirements are not met;
+``false``
+    Disable exceptions when the requirements are not met and return ``null``
+    instead;
+``null``
+    Disable checking the requirements (thus, match the route even when the
+    requirements don't match).
```

Comment:
The wording in the note is still not right - since it talks about "matching". How about:

```rst
Disable checking the requirements (thus, generate the URL even if one
or more wildcards don't match their requirements).
```